src/geobr.jl

```diff
@@ -0,0 +1,640 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+Provides functions to download data from the GeoBR database:
+
+`GeoBR.state`
+`GeoBR.municipality`
+`GeoBR.region`
+`GeoBR.country`
+`GeoBR.amazon`
+`GeoBR.biomes`
+`GeoBR.disasterriskarea`
+`GeoBR.healthfacilities`
+`GeoBR.indigenousland`
+`GeoBR.metroarea`
+`GeoBR.neighborhood`
+`GeoBR.urbanarea`
+`GeoBR.weightingarea`
+`GeoBR.mesoregion`
+`GeoBR.microregion`
+`GeoBR.intermediateregion`
+`GeoBR.immediateregion`
+`GeoBR.municipalseat`
+`GeoBR.censustract`
+`GeoBR.statisticalgrid`
+`GeoBR.conservationunits`
+`GeoBR.semiarid`
+`GeoBR.schools`
+`GeoBR.comparableareas`
+`GeoBR.urbanconcentrations`
+`GeoBR.poparrangements`
+`GeoBR.healthregion`
+"""
+module GeoBR
+
+using GeoIO
+using DataDeps
+using CSV
+using Tables
+using TableTransforms
+
+const APIVERSIONS = (v"1.7.0",)
+
+"""
+    metadata()
+
+Retrieve the metadata file path for the GeoBR project.
+
+# Returns
+The file path of the metadata CSV file.
+"""
+function metadata()
+    url = "http://www.ipea.gov.br/geobr/metadata/metadata_1.7.0_gpkg.csv"
+    ID = "GeoBR_metadata"
+    try
+        @datadep_str filepath(ID, url)
+    catch
+        register(DataDep(ID,
+            """
+            Metadata for GeoBR project.
+            Source: $url
+            """,
+            url,
+            Any
+        ))
+        @datadep_str filepath(ID, url)
+    end
+end
+
+"""
+    metadatarows(geo, year, code, abbrev; all=false)
+
+Retrieve metadata rows for the specified parameters.
+
+# Arguments
+- `geo`: The geographic level (e.g., "state", "municipality").
+- `year`: The year of the data (e.g., 2010).
+- `code`: The numeric code of the geographic area (optional).
+- `abbrev`: The abbreviation of the geographic area (optional).
+- `all`: If true, return all matching rows; if false, return only the first row (default: false).
+
+# Returns
+A single row or all rows of metadata matching the specified criteria.
+
+# Throws
+- `ErrorException` if no matching rows are found.
+"""
+function metadatarows(geo, year, code, abbrev; all=false)
+    table = CSV.File(metadata())
```

Comment:
```suggestion
    table = CSV.File(joinpath(metadata(), "metadata.csv"))
```

the `metadata()` function will just return the path to the metadata dataset, which is a directory. You can append the actual filename explicitly given that it is fixed and known.